lib/mixins/property-effects.html

```diff
@@ -1933,13 +1933,33 @@
         // Normalize fancy native splice handling of crazy start values
         if (start < 0) {
           start = array.length - Math.floor(-start);
-        } else {
+        } else if (start) {
           start = Math.floor(start);
         }
-        if (!start) {
-          start = 0;
+        // array.splice does different things based on the number of arguments
+        // you pass in. Therefore, array.splice(0) and array.splice(0, undefined)
+        // do different things. In the former, the whole array is cleared. In the
+        // latter, no items are removed.
+        // This means that we need to detect whether 1. one of the arguments
+        // is actually passed in and then 2. determine how many arguments
+        // we should pass on to the native array.splice
+        //
+        let ret;
+        // Omit any additional arguments if they were not passed in
+        if (start !== undefined && !deleteCount && !items.length) {
```

Comment:
For simplicity, let's just check if `arguments.length ==2`

Reply:
Fixed :+1: